src/iop/demosaic.c

```diff
@@ -1200,6 +1200,32 @@ static void vng_interpolate(float *out, const float *const in, const dt_iop_roi_
     for(int i = 0; i < height * width; i++) out[i * 4 + 1] = (out[i * 4 + 1] + out[i * 4 + 3]) / 2.0f;
 }
 
+/** 1:1 "demosaic" where in comes RGB buffer and out goes R=G=B=average(R,G,G,B) */
+static void rgb_monochrome(float *out, const float *const in, const dt_iop_roi_t *const roi_out,
+                           const dt_iop_roi_t *const roi_in)
+{
+  // we should always process buffers of exactly the same size
+  assert(roi_in->width == roi_out->width);
+  assert(roi_in->height == roi_out->height);
+
+#ifdef _OPENMP
+#pragma omp parallel for default(none) shared(out) schedule(static)
```

Comment:
I'll have to read up about it, I just copied that line from an equivalent loop without understanding it.